cmd/testperformance.go

```diff
@@ -187,102 +217,241 @@ func testPerformance(ctx context.Context, queuedTests []testCaseName, allTests m
 	}
 }
 
-func performanceDiskWriteTest(ctx context.Context, conf *testPerformanceConfig) testResult {
-	testRes := testResult{Name: "DiskWrite"}
-
-	log.Info(ctx, "Testing disk write...",
-		z.Any("file size MB", conf.DiskWriteMB),
-		z.Any("loops", diskWriteLoops))
-
-	if conf.DiskWriteMB <= 3072 {
-		log.Warn(ctx, "File size used for tests of 3072MB or lower impacts the measured performance", nil)
+func fioCommand(ctx context.Context, filename string, fileSize int, blocksize int, operation string) ([]byte, error) {
+	//nolint:gosec
+	cmd, err := exec.CommandContext(ctx, "fio",
+		"--name=fioTest",
+		fmt.Sprintf("--filename=%v", filename),
+		fmt.Sprintf("--size=%vMb", fileSize),
+		fmt.Sprintf("--blocksize=%vk", blocksize),
+		fmt.Sprintf("--numjobs=%v", diskOpsNumOfJobs),
+		fmt.Sprintf("--rw=%v", operation),
+		"--direct=1",
+		"--runtime=60s",
+		"--group_reporting",
+		"--output-format=json",
+	).Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "exec fio command")
 	}
 
+	return cmd, nil
+}
+
+func setupDiskOperation(ctx context.Context, conf *testPerformanceConfig) (int, string, error) {
 	var stat unix.Statfs_t
-	wd, _ := os.UserHomeDir()
-	err := unix.Statfs(wd, &stat)
+	var dir string
+	var err error
+	if conf.DiskIOTestFileDir == "" {
+		dir, err = os.UserHomeDir()
+		if err != nil {
+			return 0, "", errors.Wrap(err, "get user home dir")
+		}
+	} else {
+		dir = conf.DiskIOTestFileDir
+	}
+	err = unix.Statfs(dir, &stat)
 	if err != nil {
-		return failedTestResult(testRes, err)
+		return 0, "", errors.Wrap(err, "unix stat free storage")
 	}
-	// Available blocks * size per block = available space in bytes; remove 20% for safety
+
+	// Available blocks * size per block = available space in bytes; convert to MBs; use only 80% for safety
 	availableMB := int(stat.Bavail*uint64(stat.Bsize)/1024/1024) / 5 * 4
-	actualDiskWriteMB := conf.DiskWriteMB
+	diskWriteMB := 8192
 
-	for availableMB < actualDiskWriteMB {
-		log.Warn(ctx, fmt.Sprintf("Insufficient available disk space of %vMB, reducing the test size to %vMB. Note that this might result in slower write speed", availableMB, actualDiskWriteMB), nil)
-		actualDiskWriteMB /= actualDiskWriteMB
-		if actualDiskWriteMB == 0 {
-			return failedTestResult(testRes, errors.New("insufficient available disk space", z.Str("available_space", strconv.Itoa(availableMB)+"MB")))
+	for availableMB < diskWriteMB {
+		log.Warn(ctx, fmt.Sprintf("Insufficient available disk space of %vMB, reducing the test size of %vMB to %vMB. Note that this might result in lower performance", availableMB, diskWriteMB, diskWriteMB/2), nil)
```

Comment:
I would leave this to `fio` to determine.

Reply:
I need to specify size, it is a required parameter when running `fio`. It crashes if the specified size is more than the allowed, so that's why I'm trying with lower and lower file size here.

Reply:
We should choose an appropriate size for the test that we need to run, not the other way around :D

I would go with 4GB, if a user is running this test on an unsuitable machine, It'll be apparent lol